collect_app/src/main/java/org/odk/collect/android/tasks/InstanceUploaderTask.java

```diff
@@ -490,8 +497,16 @@ private boolean uploadOneSubmission(String urlString, String id, String instance
                         outcome.mResults.put(id, fail + response.getStatusLine().getReasonPhrase()
                                 + " (" + responseCode + ") at " + urlString);
                     } else {
-                        outcome.mResults.put(id, fail + response.getStatusLine().getReasonPhrase()
-                                + " (" + responseCode + ") at " + urlString);
+                        // Custom messaging response if fail.
+                        // If Custom Server Response enabled then confirm response is valid
+                        // and set as display message, otherwise fall back to default string messaging
+                        if (messageParser != null && messageParser.isValid){
```

Comment:
`messageParser` can't ever be `null` here because it was just initialized. `isValid` should be a call on the getter. There's the same thing again below.